source/geod24/LocalRest.d

```diff
@@ -410,6 +331,61 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
 {
     static assert (!serializerInvalidReason!(S).length, serializerInvalidReason!S);
 
+    alias RestChannel = C.Channel!Variant;
+
+    /// Ask the node to shut down
+    private struct ShutdownCommand (API)
+    {
+        /// Any callback to call before the Node's destructor is called
+        void function (API) callback;
+
+        /// Whether we're restarting or really shutting down
+        bool restart;
+    }
+
+    /// Data sent by the caller
+    private struct Command
+    {
+        /// Channel to be used when responding to this Command
+        RestChannel resp_chn;
+        /// Method to call
+        string method;
+        /// Serialized arguments to the method
+        SerializedData args;
+    }
```

Comment:
With a per command response channel, you should not need an id for re-entrancy. For the exact same behaviour, connection channel should be flushed when node restarts though.

Reply:
Per command ? You mean per type or per instance ? Assuming it's per instance, it's bringing its own host of problems.

Reply:
In the current implementation, each node has a channel that it listens for commands and each command comes with a channel that the node should respond on. So it is per command, for now.

Reply:
> My instinct would be to rather move Variant outside of this function.

If you move the Variant outside of the class, wouldn't you loose the API information? This, the structs being moved, was all because of the ShutdownCommand being templated.

Reply:
> If you move the Variant outside of the class, wouldn't you loose the API information? This, the structs being moved, was all because of the ShutdownCommand being templated.

Well you can do `Variant(API)` and have an alias here.